objects.go

```diff
@@ -128,6 +141,39 @@ func (chain *Chain) writeOperation(verb verb, ctx *nftContext, writer io.Writer)
 	fmt.Fprintf(writer, "\n")
 }
 
+func parseChain(words []string) (*Chain, error) {
+	if len(words) < 5 {
+		return nil, fmt.Errorf("chain command should have at least 5 words")
+	}
+	chainObj := &Chain{
+		Name: words[4],
+	}
+	if len(words) > 6 {
+		chainStr, comment := splitComment(strings.Join(words[6:len(words)-1], " "))
+		if comment != nil {
+			chainObj.Comment = comment
+		}
+		if len(chainStr) > 0 {
+			chainWords := strings.Split(chainStr, " ")
+			if len(chainWords) < 6 {
+				return nil, fmt.Errorf("can't parse chain description: %s", chainStr)
+			}
+			chainObj.Type = (*BaseChainType)(&chainWords[1])
+			chainObj.Hook = (*BaseChainHook)(&chainWords[3])
```

Comment:
The arguments aren't required to be in any particular order. You don't necessarily have to accept them in arbitrary order, but you should at least check that `chainWords[0] == "type"` etc